kodi_addon_checker/report.py

```diff
@@ -0,0 +1,106 @@
+import inspect
+from abc import ABC
+
+PROBLEM = "ERROR"
+WARNING = "WARN"
+INFORMATION = "INFO"
+
+
+class Report(object):
+    def __init__(self, artifact_name):
+        self.artifact_name = artifact_name
+        self.problem_count = 0
+        self.warning_count = 0
+        self.information_count = 0
+        self.reports = []
+
+    def add(self, report):
+        self.reports.append(report)
+        if type(report) is Record:
+            if PROBLEM == report.log_level:
+                self.problem_count += 1
+            elif WARNING == report.log_level:
+                self.warning_count += 1
+            else:
+                self.information_count += 1
+        else:
+            self.problem_count += report.problem_count
+            self.warning_count += report.warning_count
+            self.information_count += report.information_count
+
+    def __iter__(self):
+        return iter(self.reports)
+
+
+class Record(object):
+    def __init__(self, log_level, message, start_line=-1, end_line=-1, start_char_position=-1,
```

Comment:
Not sure if that's the python way to do things, but please give these classes their own files :)

Reply:
Fixed